### PR TITLE
Use dids package and idx field in 3ID doc

### PR DESCRIPTION
**Pull request by PaulLeCam**

### CODE REVIEW

File: packages/ceramic-common/src/ceramic-api.ts [hunk 1]
```diff
@@ -23,25 +24,17 @@ export interface PinApi {
...
-    send(jsonReq: object): JsonRpc2Response;
-}
+export type { DIDProvider } from 'dids'
 
 /**
  * Describes Ceramic node API
  */
 export interface CeramicApi {
     pin: PinApi;
+    user?: DID;
```
Comment (simonovic86) [inline packages/ceramic-common/src/ceramic-api.ts:37]: do we need `user` here?

Comment (PaulLeCam) [reply packages/ceramic-common/src/ceramic-api.ts:37]: It's used in various places in IDX to check if the user is authenticated, access its `DID` to set the owner when creating documents, resolve the DID document, etc.

Comment (oed) [reply packages/ceramic-common/src/ceramic-api.ts:37]: yep, this is just a reference to the js-did instance.

File: packages/ceramic-common/package.json [hunk 2]
```diff
@@ -28,7 +28,8 @@
   "dependencies": {
     "@types/json-schema": "^7.0.5",
     "ajv": "^6.12.3",
-    "did-resolver": "^2.1.1"
+    "did-resolver": "^2.1.1",
+    "dids": "^0.2.1"
```
Comment (oed) [inline packages/ceramic-common/package.json:32]: You should only need ti import this as a devDep as only the types are being used.

**Review by oed — COMMENTED** [review]
Looks good.

File: packages/ceramic-common/src/ceramic-api.ts (hunk 1 above)
Comment (simonovic86) [reply packages/ceramic-common/src/ceramic-api.ts:37]: makes sense! cool!

**Review by simonovic86 — APPROVED** [review]

**Pull request merged**